tapfreighter/chain_porter.go

```diff
@@ -646,6 +646,36 @@ func (p *ChainPorter) storePackageAnchorTxConf(pkg *sendPackage) error {
 		)
 	}
 
+	// Now we scan through the VPacket for any burns.
+	var burns []*AssetBurn
+
+	for _, v := range pkg.VirtualPackets {
+		for _, o := range v.Outputs {
+			if !o.Asset.IsBurn() {
+				continue
+			}
+
+			assetID := o.Asset.ID()
+
+			// We prepare the burn and add it to the list.
+			b := &AssetBurn{
+				AssetID:    assetID[:],
+				Amount:     o.Amount,
+				AnchorTxid: pkg.OutboundPkg.AnchorTx.TxHash(),
+				Note:       pkg.Note,
+			}
+
+			if o.Asset.GroupKey != nil &&
+				!o.Asset.GroupKey.RawKey.IsEmpty() {
```

Comment:
This is wrong. `IsEmpty()` tests the key locator. But we don't access the `o.Asset.GroupKey.RawKey` below. So just remove this second condition.